src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -145,7 +145,8 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
     stateParams.exchangeMgr           = chip::Platform::New<Messaging::ExchangeManager>();
     stateParams.messageCounterManager = chip::Platform::New<secure_channel::MessageCounterManager>();
 
-    ReturnErrorOnFailure(stateParams.fabricTable->Init(mFabricStorage));
+    ReturnErrorOnFailure(
+        stateParams.fabricTable->Init(mFabricStorage, chip::Platform::New<ControllerFabricDelegate>(stateParams.sessionMgr)));
```

Comment:
Doesn't this leak on failure?

Reply:
Fixed by removing the pointer from the Init method